### DISCUSSION TITLE
Rest client using multipart stopped working while migrating to Quarkus 3

### DISCUSSION THREAD

**Discussion by PierreBtz** [quarkusio/quarkus#33600] [Q&A] [answered]
Hey folks,

While migrating to Quarkus 3.x, I noticed an issue with one of my RestClients. I managed to reduce this to a minimal example showing the issue, but I cannot determine whether I'm doing something wrong that used to work by chance or if I found a genuine bug. Here goes:

```
@ApplicationScoped
@RegisterRestClient(configKey = "client")
public interface Client {

    @POST
    @Path("/api")
    void call(MultipartBody data);

    class MultipartBody {

        @RestForm("file")
        @PartType(MediaType.APPLICATION_OCTET_STREAM)
        private final InputStream file;

        public MultipartBody(InputStream file) {
            this.file = file;
        }

        public InputStream getFile() {
            return file;
        }

        @Override
        public boolean equals(Object o) {
            if (this == o) return true;
            if (o == null || getClass() != o.getClass()) return false;
            MultipartBody that = (MultipartBody) o;
            try {
                return IOUtils.contentEquals(file, that.file);
            } catch (IOException e) {
                return false;
            }
        }

        @Override
        public int hashCode() {
            return Objects.hash(file);
        }
    }
}
```
Simply running the following test with Quarkus 3.x:

```
@QuarkusTest
class ClientTest {

    @Inject
    @RestClient
    Client client;

    @Test
    void aTest() {
    }

}
```

Gives:

```
Caused by: java.lang.IllegalArgumentException: Not a REST client interface: interface eu.pierrebeitz.bug.Client. No @Path annotation found on the class or any methods of the interface and no HTTP method annotations (@POST, @PUT, @GET, @HEAD, @DELETE, etc) found on any of the methods
```

Now going back to the latest 2.x release, and adapting the import s/jakarta/javax, I have a working example.

The issue is clearly linked to the usage of the multipart object for some reason, but I don't see where I'm wrong/if I'm wrong. I didn't find any issue/PR related to this either.

Repo in case someone wants to have a look: https://github.com/PierreBtz/quarkus-rest-easy-client-bug

Any ideas?

[a comment by an automated account is not shown]

**Reply by Sgitario** [reply]
Thanks for spotting! 
I confirm this is a bug that was introduced by https://github.com/quarkusio/quarkus/pull/32061.
I'm working on a fix.

**Reply by Sgitario** [reply, marked as answer]
As a workaround, you can annotate the InputStream field with `@PartFilename(...)`

**Reply by PierreBtz** [reply]
Great, thanks for the fast answer. Want me to create a bug?

**Reply by Sgitario** [reply]
I already created it: https://github.com/quarkusio/quarkus/issues/33601
